### PR TITLE
PMM-1531 Fixed tablespaces collector for MySQL 8+

### PR DESCRIPTION
**Pull request by percona-csalguero**
Jira: https://jira.percona.com/browse/PMM-1531
FB: https://github.com/Percona-Lab/pmm-submodules/pull/1538

### CODE REVIEW

File: collector/info_schema_innodb_sys_tablespaces.go [hunk 1]
```diff
@@ -104,3 +126,24 @@ func (ScrapeInfoSchemaInnodbTablespaces) Scrape(ctx context.Context, db *sql.DB,
...
+	}
+
+	var res string
+	query := "SHOW TABLES FROM information_schema LIKE 'INNODB_SYS_TABLESPACES'"
+
+	// By default, assume MySQL 5.7+
+	innodbTablespacesQuery = innodbTablespacesQueryv57
+
+	err := db.QueryRowContext(ctx, query).Scan(&res)
+	if err != nil {
```
Comment (askomorokhov) [inline collector/info_schema_innodb_sys_tablespaces.go:142]: Just wondering, why you do not check it with `SELECT VERSION();`?

File: collector/info_schema_innodb_sys_tablespaces.go [hunk 2]
```diff
@@ -104,3 +126,24 @@ func (ScrapeInfoSchemaInnodbTablespaces) Scrape(ctx context.Context, db *sql.DB,
...
+		// if INNODB_SYS_TABLESPACES doesn't exists, assume it it MySQL 8+.
+		// The table was renamed and the file_format field was removed
+		if errors.Is(err, sql.ErrNoRows) {
+			innodbTablespacesQuery = innodbTablespacesQueryv80
```
Comment (askomorokhov) [inline collector/info_schema_innodb_sys_tablespaces.go:146]: Should we return innodbTablespacesQuery?

**Review by askomorokhov — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

File: collector/info_schema_innodb_sys_tablespaces.go (hunk 2 above)
Comment (percona-csalguero) [reply collector/info_schema_innodb_sys_tablespaces.go:146]: No. It is global to fullfil the interface.

File: collector/info_schema_innodb_sys_tablespaces_test.go [hunk 3]
```diff
@@ -11,6 +11,7 @@ import (
 )
 
 func TestScrapeInfoSchemaInnodbTablespaces(t *testing.T) {
+	innodbTablespacesQuery = innodbTablespacesQueryv57
```
Comment (BupycHuk) [inline collector/info_schema_innodb_sys_tablespaces_test.go:14]: Why do you set it here? looks like https://github.com/percona/mysqld_exporter/pull/74/files#diff-2350188e094bf6cd7238c12109c42e4e3ac067cda8add719a626fe2dcde325e6R29 should set it by itself and that's what we should test, right?

File: collector/info_schema_innodb_sys_tablespaces.go [hunk 4]
```diff
@@ -54,11 +72,15 @@ func (ScrapeInfoSchemaInnodbTablespaces) Help() string {
 
 // Version of MySQL from which scraper is available.
 func (ScrapeInfoSchemaInnodbTablespaces) Version() float64 {
-	return 5.7
+	return 8.0
```
Comment (BupycHuk) [inline collector/info_schema_innodb_sys_tablespaces.go:75]: if this scraper is available only for 8.0, I don't understand why do we set innodb tablespaces query?

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: collector/info_schema_innodb_sys_tablespaces.go (hunk 4 above)
Comment (percona-csalguero) [reply collector/info_schema_innodb_sys_tablespaces.go:75]: Not only. It works with 8.0.
There is no min-max versions.

File: collector/info_schema_innodb_sys_tablespaces_test.go (hunk 3 above)
Comment (percona-csalguero) [reply collector/info_schema_innodb_sys_tablespaces_test.go:14]: I've improved the tests.

**Pull request merged**